packages/utils/src/lib/audio.ts

```diff
@@ -0,0 +1,69 @@
+import { Howl } from "howler";
+import { useCallback, useMemo, useState } from "react";
+
+/**
+ * Hook used to play an audio file using the Howl library.
+ */
+export const usePlayAudioUrl = () => {
+  const [audio, setAudio] = useState<Howl>();
+  const [audioUrl, setAudioUrl] = useState<string>();
+  const [isAudioPlaying, setIsAudioPlaying] = useState<boolean>(false);
+
+  const playPauseAudio = useCallback(
+    (src?: string) => {
+      if (!src) return;
+
+      const isCurrentSong = src === audioUrl;
+
+      // if currently selected song, stop or play and return
+      if (isCurrentSong) {
+        if (audio?.playing()) {
+          audio?.stop();
+        } else {
+          audio?.play();
+        }
+
+        return;
+      }
+
+      // if not currently selected song, stop playing
+      if (audio?.playing()) {
+        audio.stop();
+      }
+
+      // play new song
+      const newAudio = new Howl({
+        html5: true,
+        onend: () => {
+          setIsAudioPlaying(false);
+        },
+        onpause: () => {
+          setIsAudioPlaying(false);
+        },
+        onplay: (id) => {
+          setAudioUrl(src);
+          setIsAudioPlaying(true);
+        },
+        onstop: () => {
+          setIsAudioPlaying(false);
+        },
+        src,
+      });
```

Comment:
Just a heads up, ran into issues with playback in my progress bar branch,  if the source is going to be the same as the nft Asset previews/clips like pool.pm through nftcdn, then we will need to add a `format` prop as they don't have filetype endings, which howl doesn't appreciate. I believe we only support flac and wav, not sure if the clip song preview will be in a different format.

`format: ["wav", "flac"],`

[Source of Information](https://www.npmjs.com/package/howler#src-arraystring--required:~:text=If%20your%20files%20have%20no%20extensions%2C%20you%20will%20need%20to%20explicitly%20specify%20the%20extension%20using%20the%20format%20property.)